MaterialLib/MPL/Properties/RelativePermeability/RelPermNonWettingVanGenuchten.cpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * Created on April 1, 2020, 2:15 PM
+ */
+
+#include "RelPermNonWettingVanGenuchten.h"
+
+#include <algorithm>
+#include <cmath>
+
+#include "MaterialLib/MPL/Medium.h"
+
+namespace MaterialPropertyLib
+{
+RelPermNonWettingVanGenuchten::RelPermNonWettingVanGenuchten(
+    const double Snr, const double Snmax, const double m, const double krel_min)
+    : _saturation_r(1. - Snmax),
+      _saturation_max(1. - Snr),
+      _m(m),
+      _krel_min(krel_min)
+{
+    if (!(_m > 0 && _m < 1))
+    {
+        OGS_FATAL(
+            "The exponent value m = {:g} of van Genuchten relative "
+            "permeability model, is out of its range of (0, 1)",
+            _m);
+    }
+}
+
+PropertyDataType RelPermNonWettingVanGenuchten::value(
+    VariableArray const& variable_array,
+    ParameterLib::SpatialPosition const& /*pos*/, double const /*t*/,
+    double const /*dt*/) const
+{
+    const double S_l = std::get<double>(
+        variable_array[static_cast<int>(Variable::liquid_saturation)]);
+    const double S = std::clamp(S_l, _saturation_r + _minor_offset,
+                                _saturation_max - _minor_offset);
+    const double Se = (S - _saturation_r) / (_saturation_max - _saturation_r);
+    const double krel =
+        std::cbrt(1.0 - Se) * std::pow(1.0 - std::pow(Se, 1.0 / _m), 2.0 * _m);
```

Comment:
should be sqrt?

Reply:
It is slightly change of the van Genuchten Maulem model. Now this PR was changed for  van Genuchten Maulem model, which sqrt is used.